.github/workflows/automate-terraform.yml

```diff
@@ -0,0 +1,87 @@
+name: 'Terraform GitHub Actions'
+on:
+  push:
+    branch: [wip_terraform]
+jobs:
+  backend:
+      name: Build SringbootApplication
+      runs-on: ubuntu-latest
+
+      steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: maven
+      - name: Build with Maven
+        run: mvn clean install
+      - name: Build & push Docker image
+        uses: mr-smithers-excellent/docker-build-push@v5
+        with:
+          image: yashuyellapu/employee-portal-fullstack
+          tags: backend-latest
+          registry: docker.io
+          dockerfile: Dockerfile
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+  frontend:
+        name: Build ReactJs
+        runs-on: ubuntu-latest
+
+        steps:
+          - uses: actions/checkout@v3
+          - name: Use Node.js
+            uses: actions/setup-node@v3
+            with:
+              node-version: 16
+              cache: 'npm'
+              cache-dependency-path: ./frontend/package-lock.json
+          - name: Install node packages
+            working-directory: ./frontend
+            run: |
+              npm install
+              pwd
+          - name: Build & push Docker image
+            uses: mr-smithers-excellent/docker-build-push@v5
```

Comment:
I think there's an official docker action that you can use instead of this custom one.